platform/include/platform/mbed_retarget.h

```diff
@@ -63,17 +67,36 @@ typedef unsigned int  gid_t;    ///< Group ID
 /* Flags for open() and fcntl(GETFL/SETFL)
  * At present, fcntl only supports reading and writing O_NONBLOCK
  */
+#ifndef O_RDONLY
 #define O_RDONLY 0        ///< Open for reading
+#endif
```

Comment:
Which one was redefined ? That could be terrible if some are equals to others.

Reply:
Mmm, it is worth sanity checking you haven't made a mismatched set.

The same `switch` trick for all the `O_` defines is basically sufficient. You might want to also add

    static_assert((O_ACCMODE & (O_NONBLOCK|O_APPEND|O_CREAT|O_TRUNC|O_EXCL|O_BINARY)) == 0, "O_ACCMODE clash")